php/thankYou.php

```diff
@@ -122,45 +305,64 @@
-                    echo "<p>None selected</p>";
-                }
-                else {
-                    echo "<ul>";
-                    foreach($_POST["interests"] as $interests) {
-                        if($interests === "other"){
-                            echo "<li>Other Skills: ".$_POST["interestsText"]."</li>";
-                            $mailInterests .= ", {$_POST["interestsText"]}";
-                        }
-                        else {
-                            echo "<li>" . $interests . "</li>";
-                            $mailInterests .=  ", $interests";
+                if(validSelectWText($interstArray,$_POST["interests"])) {
+                    if (!isset($_POST["interests"])) {
+                        echo "<p>None selected</p>";
+                    } else {
+                        echo "<ul>";
+                        foreach ($_POST["interests"] as $interests) {
+                            if ($interests === "other" && validText($_POST["interestsText"])) {
+                                echo "<li>Other Skills: " . $_POST["interestsText"] . "</li>";
+                                $mailInterests .= ", {$_POST["interestsText"]}";
+                            }
+                            if ($interests === "other" && validText($_POST["interestsText"]) && trim($_POST["interestsText"]) === "" ) {
+                                echo "<li>Other</li>";
+                                $mailInterests .= ", {$_POST["interestsText"]}";
+                            }
+                            else {
+                                echo "<li>" . $interests . "</li>";
+                                $mailInterests .= ", $interests";
+                            }
                         }
+                        $mailInterests = mysqli_real_escape_string($cnxn,$mailInterests);
+                        echo "</ul>";
                     }
-                    echo "</ul>";
+                }
+                else{
+                    echo "<p>Illegal Selection</p>";
+                    $isValid = false;
                 }
 
                 //Availability
-                echo "<h5>Availability</h5>";
-                if(!isset($_POST["availability"])){
-                    echo "<p>None selected</p>";
-                }
-                else {
-                    foreach($_POST["availability"] as $available) {
-                        if($available === "oneWeek"){
-                            echo "<p>Available for one week of Summer Camp</p>";
-                            $mailAvailable .= "Available One week for Summer Camp";
-                        }
-                        else if($available === "weekends"){
-                            echo "<p> Available: ".$_POST["weekendTimes"]."</p>";
-                            $mailAvailable .=  " Available: $available";
+                if(validSelectWText($availableArray, $_POST["availability"])) {
```

Comment:
"Notice:  Undefined index: availability..." is on the page when I submit an empty form while debugging.